src/Button/Button.component.tsx

```diff
@@ -3,113 +3,619 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { darken } from 'polished';
+import { transparentize } from 'polished';
 // THEME
 import { colors, fonts, sizes } from '../theme';
+import { Colors } from '../Typography/Typography.component';
+
+type Variant = 'primary' | 'outline' | 'minimal';
+
+type ButtonSize = 'xlarge' | 'large' | 'small' | 'xsmall';
+
+interface Theme {
+    base: string;
+    light: string;
+    dark: string;
+}
 
 interface ButtonProps {
     children?: any;
     disabled?: boolean;
+    revealed?: boolean;
     showSpinner?: boolean;
     className?: string;
 
+    flip?: boolean;
+    colorTheme?: Theme | Colors;
+    minWidth?: string;
+    block?: boolean;
+    icon?: any;
+    circular?: boolean;
+    reverse?: boolean;
+
+    forceHover?: boolean;
+    forceFocus?: boolean;
+    forceActive?: boolean;
+
     onClick?: () => void;
-    variant?: 'primary' | 'secondary' | 'text';
-    size?: 'small' | 'medium' | 'large';
+    variant?: Variant;
+    size?: ButtonSize;
 }
 
-interface ButtonStylesGroup {
-    [key: string]: FlattenSimpleInterpolation;
+interface StyledButtonProps extends ButtonProps {
+    iconOnly?: boolean;
+    height: number;
+    borderRadius: string;
+
+    buttonStyles: ButtonStyles;
+
+    // make nonnullable
+    colorTheme: Theme;
+    variant: Variant;
+    size: ButtonSize;
 }
 
-const ButtonColorStyles: ButtonStylesGroup = {
-    primary: css`
-        border: solid thin ${colors.savvyCyan.base};
-        background-color: ${colors.savvyCyan.base};
-        color: ${colors.white.base};
+const themeDefaults = {
```

Comment:
I don't love this implementation for colors (See [here](https://github.com/RetailMeNot/anchor/pull/45/files#diff-2e2fd1afd183724a048e4d14f7ea5e3fR580)). It maps really well for the primary button, but not all of the variants and reverse variants fit well into this paradigm, with only 1 or 2 of the hues actually being used or some of the hues not being applied cleanly.

For instance, `minimal - primary` only makes use of savvyCyan.dark, but `minimal - secondary` (which is not included in this PR) replaces savvyCyan **dark** with charcoal **light** and also includes a hover color (charcoal.base).

I do like that we aren't totally coupled to our own themes (you can provide your own {base, light, dark}).

Reply:
🤔

Reply:
Based on our conversation I say push it out. This method is tightly integrated into the component as it stands.